internal/parsing/javascript_cucumber_json_parser.go

```diff
@@ -0,0 +1,167 @@
+package parsing
+
+import (
+	"encoding/json"
+	"io"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/rwx-research/captain-cli/internal/errors"
+	v1 "github.com/rwx-research/captain-cli/internal/testingschema/v1"
+)
+
+type JavaScriptCucumberJSONParser struct{}
+
+type JavaScriptCucumberJSONFeature struct {
+	URI         string `json:"uri"`
+	ID          string `json:"id"`
+	Keyword     string `json:"keyword"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Line        int    `json:"line"`
+	Tags        []struct {
+		Name string `json:"name"`
+		Line *int   `json:"line"`
+	} `json:"tags"`
+	Elements []struct {
+		ID          string `json:"id"`
+		Keyword     string `json:"keyword"`
+		Name        string `json:"name"`
+		Description string `json:"description"`
+		Line        int    `json:"line"`
+		Steps       []struct {
+			Arguments []struct{} `json:"arguments"`
+			Keyword   string     `json:"keyword"`
+			Line      int        `json:"line"`
+			Name      string     `json:"name"`
+			Match     struct {
+				Location string `json:"location"`
+			} `json:"match"`
+			Result *struct {
+				Status       string `json:"status"`
+				Duration     *int   `json:"duration"`
+				ErrorMessage string `json:"error_message"`
+			} `json:"result"`
+		} `json:"steps"`
+		Tags []struct {
+			Name string `json:"name"`
+			Line *int   `json:"line"`
+		} `json:"tags"`
+		Type string `json:"type"`
+	} `json:"elements"`
+}
+
+func (p JavaScriptCucumberJSONParser) Parse(data io.Reader) (*v1.TestResults, error) {
```

Comment:
This is mostly a copy of the logic in Ruby Cucumber, but a slight tweak to the before/after hook logic which ends up in `Steps` in the JS version instead of a separate field in the struct